### DISCUSSION TITLE
Request for help

### DISCUSSION THREAD

**Discussion by akprasad** [ambuda-org/vidyut#139] [General]
Follow-up to https://groups.google.com/g/sanskrit-programmers/c/zoZ4-9eitSY/m/GctcCh9nAQAJ .

If you are interested, please reply here.

**Comment by linguistrix** [comment]
Hi Arun, happy to brainstorm ways to incorporate this technology into SanskritSahitya.org

**Reply by akprasad** [reply]
[vidyut-prakriya](https://github.com/ambuda-org/vidyut/tree/main/vidyut-prakriya) performs the semantics &rarr; word mapping, and [vidyut-kosha](https://github.com/ambuda-org/vidyut/tree/main/vidyut-kosha) performs the word &rarr; semantics mapping. 

For your use case, the second mapping is probably more useful. Where do you have a word and want to show information about that word? Also, what languages do you use for your site? Vidyut has bindings for WebAssembly and has Python bindings in progress, but I can support others as well.

**Reply by linguistrix** [reply]
Currently everything is client-side JS. Data is pulled from github / CDN and cached locally, so everything can run without an internet connection once loaded. 

I currently have DCS annotations for whatever text they were available for. Example:
![image](https://github.com/user-attachments/assets/35c5568e-477a-4027-b483-7db203ade813)

https://sanskritsahitya.org/kumarasambhavam/5.6 (You can toggle the पदच्छेदः on and off)

I am also trying to incorporate the tagging work done by Sebastian Nehrdich. 

So one option could be to have the mapping be shown for each word powered by your tool.

**Reply by akprasad** [reply]
To support this client-side we need the kosha to be much smaller, as it's currently about 50MB when zipped. This is feasible but needs some engineering work. I think 10MB is a good upper limit to aim for.

Adding @neeleshb since he and I have talked about something similar.

**Comment by chandrasekharanr** [comment]
Hi Arun, I can help.. I am not a programmer by profession but I can call myself a beginner - intermediate level Python programmer. I read Kashika and Siddhanta kaumudi vrttis (as and when required) for my self-study of vyAkaraNam. How can get started in helping?

Regards,
Chandru

**Reply by akprasad** [reply]
Here's an important project I'd love help with: find words that `vidyut.kosha` does not understand so that we can add them to our test cases and improve our output. This would help immensely. It's a very high impact project that only needs a bit of time and some basic knowledge of Python and vyAkaraNam.

Here's the basic idea:

- You can get the full text of the SK here: https://github.com/ashtadhyayi-com/data/blob/master/sutraani/kaumudi.txt
- Use [vidyut.lipi](https://vidyut.readthedocs.io/en/latest/lipi.html) or your favorite transliterator to convert this text to SLP1.
- Split the text into separate padas. Items like तत्रादौ (SLP1 `tatrAdO`) are hard to separate, but that's fine; for our purposes, we can treat them like a single pada, as we can filter them out later.
- Use [vidyut.kosha](https://vidyut.readthedocs.io/en/latest/kosha.html) to see if each pada is in the kosha. If it isn't, print out the pada and the sutra number it comes from.
- Output should be a CSV with two columns: the SK sutra number, and the "pada" that could not be found. You can use the bulit-in [csv](https://docs.python.org/3/library/csv.html) library to handle this.

Once we have this CSV, we can turn it into a spreadsheet and share it with volunteers to mark which are real errors and which are just noise.

Tips:

- Focus on chapters 8 - 13 (inclusive) for subantas and 43 - 58 (inclusive) for tinantas.
- Example padas are usually offset by dandas, e.g. `। लेख्यति ।`. So if multiple words are between dandas, e.g. `। द्विधा हि कण्ड्वादयः धातवः प्रातिपदिकानि च ।`, these are probably not example padas and are likely just noise. If you are able to, include these padas anyway, maybe with a third CSV column `is_probably_noisy`. I want to make sure we catch everything.

Please share your early results either here or on our [Discord server](https://discord.gg/7rGdTyWY7Z) on the #vidyut channel.

**Reply by akprasad** [reply]
Logged as an issue in #157 (also see #158, #159). Let's move discussion there.

**Comment by harivyoma** [comment]
I can help as well. It has been a while since programming. I'd love to learn Rust and Python. I am a beginner in Sanskrit; perhaps this effort might help me speed up my Sanskrit learning.

Please let me know when we can connect.
vyomavaryah@gmail.com

**Comment by akprasad** [comment]
Closing -- let's move discussion to ambuda-discuss .
